lib/route/link/bridge_info.c

```diff
@@ -779,6 +845,99 @@ int rtnl_link_bridge_get_mcast_snooping(struct rtnl_link *link, uint8_t *value)
 	return 0;
 }
 
+/**
+ * Set the no link-local learn flag
+ * @arg link	Link object of type bridge
+ * @arg value	Value to set the flag to
+ *
+ * @see rtnl_link_bridge_get_no_linklocal_learn()
+ *
+ * @return void
+ */
+void rtnl_link_bridge_set_no_linklocal_learn(struct rtnl_link *link, bool value)
+{
+	bridge_info_set_boolopt(link, BR_BOOLOPT_NO_LL_LEARN, value);
+}
+
+/**
+ * Get the no link-local learn flag
+ * @arg link	Link object of type bridge
+ * @arg value	Output argument.
+ *
+ * @see rtnl_link_bridge_set_no_linklocal_learn()
+ *
+ * @return Zero on success, otherwise a negative error code.
+ * @retval -NLE_NOATTR
+ * @retval -NLE_INVAL
+ */
+int rtnl_link_bridge_get_no_linklocal_learn(struct rtnl_link *link, bool *value)
```

Comment:
I really think that instead of adding all those `rtnl_link_bridge_get_no_linklocal_learn()` accessors, just expose `bridge_info_set_boolopt()` as public API.

Note that enum values like `BR_BOOLOPT_NO_LL_LEARN` are already public API (in kernel headers). Just let the user include `if_bridge.h` and set the flags themselves. The benefit is, that we support also future enum values.

Also, no need for the check `if (opt < 0 || opt >= BR_BOOLOPT_MAX)`, if the user sets a value that is not supported then kernel will tell us. On the other hand, we must check that opt is < 32, so that the bitshift is not undefined behavior.

Reply:
I thought that might be brought up. I initially did it that way because I thought it might be nice to have a higher level API that abstracts the boolopts detail.

I've made the boolopts setter and getter public and removed the individual functions for now.